### PR TITLE
Fix unsynchronized writes and blocking calls for proxy sender

### PR DESCRIPTION
**Pull request by hanwavefront**

### CODE REVIEW

File: Wavefront.SDK.CSharp/Wavefront.SDK.CSharp.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <PackOnBuild>true</PackOnBuild>
     <NoWarn>NU1701</NoWarn>
     <PackageId>Wavefront.SDK.CSharp</PackageId>
-    <PackageVersion>1.0</PackageVersion>
+    <PackageVersion>1.1.0-rc-001</PackageVersion>
```
Comment (vikramraman) [inline Wavefront.SDK.CSharp/Wavefront.SDK.CSharp.csproj:7]: Should this just be 1.1.0?

File: Wavefront.SDK.CSharp/Common/ReconnectingSocket.cs [hunk 2]
```diff
@@ -85,52 +133,90 @@ private void ResetSocket()
         /// </summary>
         /// <param name="message">The message to be sent to the Wavefront proxy.</param>
         public void Write(string message)
-        {
+        {
+            _ = WriteAsync(message);
+        }
+
+        private async Task WriteAsync(string message)
```
Comment (vikramraman) [inline Wavefront.SDK.CSharp/Common/ReconnectingSocket.cs:143]: Is there synchronization between ConnectAsync and WriteAsync?

File: Wavefront.SDK.CSharp/Wavefront.SDK.CSharp.csproj (hunk 1 above)
Comment (hanwavefront) [reply Wavefront.SDK.CSharp/Wavefront.SDK.CSharp.csproj:7]: Planning to update to 1.1.0 right before release.  Packages get cached locally by version number, so it's easier to work with when each change has a different version number.

File: Wavefront.SDK.CSharp/Common/ReconnectingSocket.cs (hunk 2 above)
Comment (hanwavefront) [reply Wavefront.SDK.CSharp/Common/ReconnectingSocket.cs:143]: Currently no, which means that writes get dropped when they are concurrent with connect or reset.  I tried letting WriteAsync wait for ResetSocket to finish, but that ends up blocking when there are numerous writes in succession and an unresponsive host.

**Review by vikramraman — APPROVED** [review]
Lgtm.

**Pull request merged**